services/src/test/scala/cromwell/services/metadata/impl/DeleteMetadataActorSpec.scala

```diff
@@ -1,34 +1,69 @@
-import cromwell.services.metadata.MetadataService.{DeleteMetadataAction, DeleteMetadataFailedResponse}
+import cromwell.services.metadata.impl.DeleteMetadataActor.DeleteMetadataAction
 import org.scalatest.FlatSpecLike
 
 import scala.concurrent.{ExecutionContext, Future}
 import scala.concurrent.duration._
+import scala.language.postfixOps
 
 class DeleteMetadataActorSpec extends TestKitSuite with FlatSpecLike {
 
-  private val workflowId = UUID.randomUUID().toString
-  private val deleteMetadataActor = system.actorOf(Props(new DeleteMetadataActor {
-    override def deleteNonLabelMetadataEntriesForWorkflow(rootWorkflowId: WorkflowId)(implicit ec: ExecutionContext): Future[Int] = {
-      Future.failed(new RuntimeException("Test exception"))
+  private val workflowId1 = UUID.randomUUID().toString
+  private val workflowId2 = UUID.randomUUID().toString
+
+  val probe = TestProbe()
+
+  private val deleteMetadataActor = createTestDeletionActor()
+  private val deleteMetadataActorFailingLookups = createTestDeletionActor(failLookups = true)
+  private val deleteMetadataActorFailingDeletions = createTestDeletionActor(failDeletions = true)
+
+  it should "lookup for workflows matching criteria and delete their metadata" in {
+    EventFilter.info(pattern = s"Successfully deleted metadata for workflow *", occurrences = 2) intercept {
+      probe.send(deleteMetadataActor, DeleteMetadataAction)
     }
-  }))
+  }
 
-  it should "try to repeat delete action required amount of times in case of failure" in {
-    val probe = TestProbe()
-    EventFilter.error(start = "Cannot delete metadata. Remaining number of attempts", occurrences = 4) intercept {
-      probe.send(deleteMetadataActor, DeleteMetadataAction(WorkflowId.fromString(workflowId), probe.ref, maxAttempts = 5))
+  it should "write error message to the log if lookup fails" in {
+    EventFilter.error(start = "Cannot delete metadata: unable to query list of workflow ids for metadata deletion from metadata summary table.", occurrences = 1) intercept {
+      probe.send(deleteMetadataActorFailingLookups, DeleteMetadataAction)
     }
-    probe.expectMsgPF(5.seconds) {
-      case DeleteMetadataFailedResponse(_, _) => // success
+  }
+
+  it should "write error message to the log if deletion fails and continue processing next workflow from the list" in {
+    // error message should appear twice
+    EventFilter.error(pattern = s"Cannot delete metadata for workflow *", occurrences = 2) intercept {
+      probe.send(deleteMetadataActorFailingDeletions, DeleteMetadataAction)
     }
   }
+
+  private def createTestDeletionActor(failLookups: Boolean = false, failDeletions: Boolean = false) = {
+    val deleteMetadataActor = system.actorOf(Props(new DeleteMetadataActor(1 minute) {
+      override def queryRootWorkflowSummaryEntriesByArchiveStatusAndOlderThanTimestamp(archiveStatus: Option[String], thresholdTimestamp: OffsetDateTime)(implicit ec: ExecutionContext): Future[Seq[String]] = {
```

Comment:
Can we also check that the `archiveStatus` is correct and that the requested `thresholdTimestamp` is set to (at least) a minute in the past when this method is called?